bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -329,10 +346,222 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
                 this._populateWmsMapLayerAdditionalData(layer, newLayerConf);
             }
 
+            // Also update layer to me._layerGroups
+            this.updateGroupsLayers(layerId, newLayerConf);
+
             // notify components of layer update
             var evt = Oskari.eventBuilder('MapLayerEvent')(layer.getId(), 'update');
             this._sandbox.notifyAll(evt);
         },
+        /**
+         * Dalete layer group
+         * @method deleteLayerGroup
+         * @param  {Object}         data group data
+         */
+        deleteLayerGroup: function(data) {
+            var me = this;
+            var newGroups = [];
+            if (data.parentId === -1) {
+                me._layerGroups.forEach(function(group) {
+                    if (group.id != data.id) {
+                        newGroups.push(group);
+                    }
+                });
+                me._layerGroups = newGroups;
+            } else {
+                var editable = me.getAllLayerGroups(data.parentId);
+                editable.groups.forEach(function(group) {
+                    if (group.id != data.id) {
+                        newGroups.push(group);
+                    }
+                });
+                editable.groups = newGroups;
```

Comment:
Nothing is done for the "editable" variable so this whole else block doesn't really do anything.

Reply:
Oh, it modifies the internals of the class variable returned by getAllLayerGroups(). If getAllLayerGroups(-1) would return this._layerGroups you wouldn't need the if-statement at all. Also the function can just take parentID and id instead of a data-object?